extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/DatasourceOperations.java

```diff
@@ -171,6 +181,20 @@ public boolean isAlreadyExistsException(SQLException e) {
   }
 
   private Connection borrowConnection() throws SQLException {
-    return datasource.getConnection();
+    Connection connection = datasource.getConnection();
+    if (connection
+        .getMetaData()
+        .getDatabaseProductName()
+        .toLowerCase(Locale.ROOT)
+        .equals("postgresql")) {
```

Comment:
Can we avoid execution conditional on the database name? At least could we refactor the code to use sub-classes instead of `if`s?

Reply:
i restructured it, please have a pass on what you think